presto-parquet/src/main/java/com/facebook/presto/parquet/batchreader/decoders/ValuesDecoder.java

```diff
@@ -60,6 +60,9 @@ interface Int64TimeAndTimestampMicrosValuesDecoder
         void readNext(long[] values, int offset, int length)
                 throws IOException;
 
+        void readNextWithTimezone(long[] values, int offset, int length)
+                throws IOException;
+
```

Comment:
This is necessary because the decoder is accesses the individual raw values from parquet, not the batch reader. The batch reader has the column descriptor (metadata) which should tell it whether or not to pack the value with timezone, like is done in the regular column reader. The actual packing should be done within the decoder, so there needs to be a way for decoders to dynamically switch between with and without timezone mode.

The approach I took here is just to copy the readNext method from each implementation and add `packDateTimewithZone`. A stateful approach (instance variable bool withTimestamp or something) would probably be more efficient for future development and code execution but for now I am seeing if this implementation works.